core/src/main/scala/kafka/server/BrokerToControllerChannelManagerImpl.scala

```diff
@@ -178,6 +191,10 @@ class BrokerToControllerRequestThread(networkClient: KafkaClient,
     }
   }
 
+  private def isTimedOut(request: BrokerToControllerQueueItem): Boolean = {
+    time.milliseconds() > request.deadlineMs
```

Comment:
Maybe we can avoid this call to `time.milliseconds` and use `ClientResponse.receivedTimeMs`?